### ISSUE TITLE
[Grails] Language Server Slow to Respond for Large Project

### ISSUE THREAD

**Issue by JoelProminic** [prominic/Moonshine-IDE#591]
I worked with a Prominic coworker to test Moonshine with a larger Grails project.  Unfortunately, the language server was not useable, because it took minutes for it to start working after I opened an editor.

The project will be released soon on GitHub.  I'll let you know when it is available so that you can test on that same project.  If I find another public project that can reproduce the behavior, I'll report that as well.

**Comment by joshtynjala** [comment]
Was it during the time that the `gradle eclipse` command was running in the console? I've noticed that this command seems to take a long time sometimes (minutes, like you saw). The language server cannot actually be started until this is done because it creates the *.classpath* file needed by the language server to know which classpaths to set up for code intelligence. However, once the `gradle eclipse` command has finished running once for a particular project, opening the same project again later should be significantly faster.

I don't know everything that Gradle does while this command is running, but it may be that it needs to download required dependencies.

**Comment by JoelProminic** [comment]
"gradle eclipse" finished in 27 seconds for this project on my machine, so I don't think this is the limiting factor.

When I open a class file, it takes several minutes for the tooltips to be available, but they do show up eventually.  If I open another file, there is a long delay again.

Sorry for the lack of a project to reproduce this issue.  I was hoping this would be available sooner, but I wanted to create an issue to make sure that this behavior was tracked in the meantime.

**Comment by joshtynjala** [comment]
In the SwitchBoard app that you shared with me, I imported the *interface-source* folder as a project in Moonshine.

From the Project menu, I clicked "Open/Import Project". Then, I chose the *interface-source* folder. After this, Moonshine ran the `gradle eclipse` command, which took 3 minutes 59 seconds to initialize the project. However, when I reopened the project a second time by choosing it from Recent Projects in Moonshine, `gradle eclipse` finished in just 3 seconds.

In both cases, as soon as `gradle eclipse` completed, I was able to open a *.groovy* file and immediately get code intelligence. No long delays at all. For both hover and completion, everything seemed instantaneous.

Steps I followed after `gradle eclipse` completed:

1. Open *grails-app/controllers/switchboard/cloud/AdminController.groovy*

2. After this line, press Enter:
    `Agent user = getAuthenticatedUser() as Agent`

3. Type `user.` and the completion list will appear immediately. I see `accountExpired`, `accountLocked`, etc.

**Comment by joshtynjala** [comment]
I cannot reproduce on macOS 10.14.6 either. The `gradle eclipse` command took 1 minute 18 seconds the first time, and only 1 second the second time. Similarly, there was no delay on completion.

**Comment by JoelProminic** [comment]
Sorry @joshtynjala, when I created this issue I expected to have a public example project available within a week or so.   This project is still stuck as a private project on GitHub.

However, I retested this project from GitHub (this copy has been modified from the original copy I tested), and I see that the reported behavior now longer triggers.   The Language Server started for me in 55 seconds, and I didn't notice any delay before the language features started working when opening a Groovy class, constructor, or domain.

I retried the old SVN project that was having problems, and I found that it was actually even faster.  "gradle eclipse" finished in 5 seconds, and there was again no noticeable delay after opening the files.

I'm going to close the issue for now.  I'll try to use this more for any Grails development I do, and I'll let you know if I see this behavior again.
